app/controllers/proposals_controller.rb

```diff
@@ -150,27 +149,15 @@ def require_waitlisted_or_accepted_state
     end
   end
 
-  def require_complete_profile
-    return if current_user.complete? && current_user.unconfirmed_email.blank?
-
-    @incomplete_profile_msg = construct_incomplete_profile_msg.html_safe
-  end
-
-  def construct_incomplete_profile_msg
-    base_msg = "Before submitting a proposal your profile needs completing. Please correct the following:"
-
-    confirm_email_msg = " The email #{current_user.unconfirmed_email} needs confirmation."
-    base_msg << confirm_email_msg if current_user.unconfirmed_email.present?
-
-    profile_link_msg = " Visit #{view_context.link_to('My Profile', edit_profile_path)} and fix error(s):"
-    base_msg << profile_link_msg if current_user.name.blank? || include_missing_email_message?(base_msg, confirm_email_msg)
-
-    base_msg << " Name can't be blank." if current_user.name.blank?
-    base_msg << " Email can't be blank (confirmation required)." if include_missing_email_message?(base_msg, confirm_email_msg)
-    base_msg
-  end
-
-  def include_missing_email_message?(base_msg, confirm_email_msg)
-    current_user.email.blank? && !base_msg.include?(confirm_email_msg)
+  def incomplete_profile_msg
+    if profile_errors = current_user.profile_errors
+      err_msgs = profile_errors.messages
+      msg = "Before submitting a proposal your profile needs completing. Please correct the following: "
+      msg << profile_errors.full_messages.to_sentence
+      if err_msgs[:name] || err_msgs[:email][0].include?('blank')
```

Comment:
Still not 100% here style/syntax, but the idea is that I don't want to direct the user with a link to 'My Profile' for an `:unconfirmed_email` b/c that could mislead them into thinking visiting that page would resolve it. I also don't want to include the link if both `:email` is blank and `:unconfirmed_email` is present, b/c confirming will resolve blank `:email`. I'm fine including the link if `:name` is blank and `:unconfirmed_email` is present.